editor/src/components/canvas/controls/breadcrumb-trail.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react'
+import { useEditorState } from '../../../components/editor/store/store-hook'
+import {
+  betterReactMemo,
+  useKeepReferenceEqualityIfPossible,
+} from '../../../utils/react-performance'
+import { Utils } from '../../../uuiui-deps'
+import * as TP from '../../../core/shared/template-path'
+import { getOpenUtopiaJSXComponentsFromState } from '../../editor/store/editor-state'
+import { MetadataUtils } from '../../../core/model/element-metadata-utils'
+import { selectComponents } from '../../../components/editor/actions/action-creators'
+import { FlexRow, Icons } from '../../../uuiui'
+import { TemplatePath } from 'src/core/shared/project-file-types'
+
+interface ElementPathElement {
+  name?: string
+  path: TemplatePath
+}
+
+const { dispatch, jsxMetadataKILLME, rootComponents } = useEditorState((store) => {
```

Comment:
These React hooks should be inside a React component, not at the root level of the file.